bitrisescript/src/bitrisescript/bitrise.py

```diff
@@ -0,0 +1,218 @@
+import asyncio
+import logging
+import os
+from pprint import pformat
+from typing import Any
+
+from aiohttp_retry import RetryClient
+
+from scriptworker_client.exceptions import TaskVerificationError
+from scriptworker_client.utils import get_single_item_from_sequence
+
+BITRISE_API_URL = "https://api.bitrise.io/v0.1"
+
+log = logging.getLogger(__name__)
+
+
+class BitriseClient(RetryClient):
+    """A singleton to facilite sharing client across requests."""
+
+    _instance = None
+    base: str = BITRISE_API_URL
+    prefix: str = ""
+    headers: dict = {}
+
+    def __new__(cls):
+        if cls._instance is None:
+            cls._instance = super().__new__(cls)
+        return cls._instance
+
+    def set_auth(self, token):
+        """Sets the authorization header to the provided Bitrise access token.
+
+        Args:
+            token (str): Bitrise API token.
+        """
+        self.headers["Authorization"] = token
+
+    async def set_app_prefix(self, app):
+        """Sets the client's prefix to the specified Bitrise app.
+
+        This scopes the client to only the specified application.
+
+        Args:
+            app (str): Name of the Bitrise application to scope to.
+        """
+        apps = (await self.do_request("/apps"))["data"]
+        app = get_single_item_from_sequence(
```

Comment:
suggestion: use a distinct name for the function argument and this return value. If I understand correctly, `app` starts off as a string when it comes into the function, and then turns into a dict after this returns.